[TIMOB-25748] iOS: Remove TiCore in favor of built-in JavaScriptCore

hansemannn opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25748

⚠️ Do not merge, yet! We need to wait until master is 8.0.0.

`no tests` as there are no new API's, but all existing test-suites should pass.

hansemannn committed:
Remove TiCore

hansemannn committed:
Remove Ti -> JS aliases, remove TiCore references in the CLI, unlink libTiCore

hansemannn committed:
Fix sorted imports

build commented:
<!--
  2 failure:  :disappointed_rel..., :microscope: Ther...
  0 warning: 
  2 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/hansemannn/titanium_mobile/blob/remove-ticore/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :thumbsup: Hey!, You deleted more code than you added. That's awesome!
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.1.0 test /Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA
> grunt

Running "appcJs:src" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA/build/ios.js
  12:2   warning  'downloadURL' is assigned a value but never used        no-unused-vars
  16:2   warning  'TI_CORE_VERSION' is assigned a value but never used    no-unused-vars
  17:2   warning  'TI_CORE_INTEGRITY' is assigned a value but never used  no-unused-vars
  19:10  warning  'gunzip' is defined but never used                      no-unused-vars

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA/android/cli/commands/_build.js
  1638:9  warning  Blocks are nested too deeply (6)  max-depth

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA/android/cli/commands/_buildModule.js
  554:18  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA/iphone/cli/commands/_build.js
  4241:66  error  Parsing error: Unexpected token )

✖ 7 problems (1 error, 6 warnings)

Warning: Task "eslint:src" failed. Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

hansemannn committed:
Fix typo

hansemannn committed:
Another typo

hansemannn committed:
Merge branch 'master' into remove-ticore

sgtcoolguy commented on build/ios.js:
```diff
@@ -43,35 +43,9 @@ IOS.prototype.clean = function (next) {
-
 IOS.prototype.build = function (next) {
-	this.fetchLibTiCore(next);
+	// no-op (used to fetch TiCore in the past)
```
Clearly not related to this PR, but this jogs my memory. Weren't we going to pre-build libraries for iOS like we do for the other platforms at some point? Do we have JIRA tickets/PRs to track that?

hansemannn replied:
I remember you had an angry TODO comment somewhere that states that we should just place libraries inside the repo instead of fetching them manually. In case you mean that, it should be obsolete now that we remove TiCore. Other libraries like libAPSHTTPClient and libAPSAnalytics are prepackaged, just like Hyperloop should / will be in the future.

sgtcoolguy commented on build/ios.js:
```diff
@@ -80,7 +54,6 @@ IOS.prototype.package = function (packager, next) {
 	const DEST_IOS = path.join(packager.zipSDKDir, 'iphone');
 
 	async.parallel([
-		this.fetchLibTiCore.bind(this),
```
Also need to remove the step below where we try to copy iphone/headers/JavaScriptCore since there's no files there anymore (that's why the build is failing right now, lines 62-64 in your modified file) - and the 'headers' entry in the array of folders/files we try to copy (line 66 in your modified file)

sgtcoolguy replied:
Can we also remove libti_ios_debugger.a? libti_ios_profiler.a?

Those were to be used against TiCore and do not work with JSCore.

I've been told several times we can kill off our profiler because it's not useful and no one uses/knows about it. The debugger has been replace by using the official debugger APIs built into JSCore.

hansemannn committed:
Remove TiProfiler and TiDebugger, cleanup build scripts

hansemannn committed:
Merge branch 'master' into remove-ticore

hansemannn committed:
Merge branch 'master' into remove-ticore

sgtcoolguy committed:
Merge branch 'master' into remove-ticore

sgtcoolguy committed:
Fix so Ti.UI.TextArea.editable defaults to true (matching our docs).

sgtcoolguy committed:
Don't try to copy headers folder that no longer exists. Clean up unused code in iOS build script.

sgtcoolguy committed:
Merge branch 'master' into remove-ticore

sgtcoolguy committed:
Fix indentation, remove unused 'humanize' dependency

hansemannn committed:
Merge branch 'master' into remove-ticore

hansemannn committed:
Resolve remaining merge conflicts

hansemannn committed:
Linting

hansemannn commented:
@sgtcoolguy Ready for re-review

build commented:
<!--
  1 failure:  :disappointed_rel...
  0 warning: 
  2 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :thumbsup: Hey!, You deleted more code than you added. That's awesome!
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.4.0 test /Users/build/jenkins/workspace/sdk_titanium_mobile_PR-9798-44VE7Z6T3SLF4LFJLA4UEOH4MOHHPG5QJR4XEES7IT74LU6FPXUA
> grunt

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

Running "checkFormat:ios" (checkFormat) task
Fatal error: Error: Formatting incorrect on "iphone/Classes/PlatformModule.m", proposed changes: <?xml version='1.0'?>
<replacements xml:space='preserve' incomplete_format='false'>
<replacement offset='8658' length='0'> </replacement>
</replacements>

Error: Formatting incorrect on "iphone/Classes/TiDOMDocumentProxy.m", proposed changes: <?xml version='1.0'?>
<replacements xml:space='preserve' incomplete_format='false'>
<replacement offset='14515' length='0'> </replacement>
<replacement offset='14517' length='0'> </replacement>
</replacements>

npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

hansemannn committed:
Update PlatformModule.m

hansemannn committed:
Update TiDOMDocumentProxy.m

sgtcoolguy committed:
Merge branch 'master' into remove-ticore

sgtcoolguy committed:
Fix formatting

hansemannn committed:
Use Hyperloop 3.1.0

jquick-axway committed:
[TIMOB-26055] Android: Fixed bug where CLI would only replace 1st instance of "${applicationId}" in AAR manifest. (#10206)

- Now replaces all ${applicationId} instances in AAR AndroidManfest.xml.
- Note: This was a bug when building with module "titanium-firebase-core" via Titanium 7.3.0.

drauggres committed:
[TIMOB-26204] Android: Fix transformAfterLayout NPE (#10202)

hansemannn committed:
[TIMOB-26211] Round percentage up instead of down (parity) (#10194)

sgtcoolguy committed:
Migrate tests to suite

sgtcoolguy committed:
Bump to 8.0.0, publish 'next' branch.

hansemannn commented on iphone/cli/commands/_build.js:
```diff
@@ -5838,10 +5799,6 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
 								};
-								// generate our transpile target based on tijscore/jscore
-								if (this.useJSCore) {
-									analyzeOptions.targets = { ios: this.minSupportedIosSdk }; // if using jscore, target our min ios version
```
Dangerous issue, thanks for picking this up!

sgtcoolguy commented on iphone/headers/JavaScriptCore/TiBase.h:
```diff
@@ -1,155 +0,0 @@
-/**
```
So to follow on:

I was mistaken and thought TiBase.h was removed in this PR, but it is not.

However, The native module template uses TiModule.h, which tracing up the stack imports KrollObject.h which swapped from ToTiJS.h to the actual JavaScriptCore.h.

This means that any aliased Ti* types are now gone. These types were actually available to native modules (though I'm not sure how many actually used them directly). As a result, this is a breaking api change for modules, so the api version for iOS modules should be bumped: https://github.com/appcelerator/titanium_mobile/blob/master/package.json#L6

That would mean that native modules would need to be rebuilt/re-released. I'm not sure if they were going to need to be anyways as a result of this change...

hansemannn replied:
Did we test that so far? I would reaaaally like to prevent a breaking module change wherever possible. One possible solution could be to leave the bridge file where it is (legacy), but add a deprecation warning for developers to migrate them. I am planning the same for the Swift module effort, which would basically have a "shelled" file that only imports existing API's. This could eventually work out. Should I try that in this PR, a different one or should we break the modules intentionally?

hansemannn replied:
@sgtcoolguy I just tested with an existing app (that uses Ti.Facebook v3.8.0) and it works without recompiling the module. I think that unless we actively use TiCore symbols in modules (like `TiValueRef`), we are fine. And since those are supposed to be private API anyway, people should not use it.

hansemannn committed:
Restore package-lock.json, restore iOS CLI option
